chall07/migferna.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+func rgbToHexadecimal(response string) (int, int, int, int) {
+	str := strings.Split(response, " ")
+	data := str[0]
+	values := strings.Split(data, ",")
+	id, error := strconv.Atoi(strings.Split(values[0], "=")[1])
+	red, error := strconv.Atoi(strings.Split(values[1], "=")[1])
+	green, error := strconv.Atoi(strings.Split(values[2], "=")[1])
+	blue, error := strconv.Atoi(strings.Split(values[3], "=")[1])
+
+	if error != nil {
+		log.Fatalln(error)
+	}
```

Comment:
A most simple way of doing this would have been by using [fmt.Sscanf](https://golang.org/pkg/fmt/#Sscanf). As you know exactly how the input should be:
```suggestion
	var id, red, green, blue int
	if _, err := fmt.Sscanf(response, "id=%d,r=%d,g=%d,b=%d - Send your response here: chall07.42madrid.com/?id=<id>&resp=<hex>", &id, &red, &green, &blue); err != nil {
		log.Fatalln(err)
	}